File: problems/tier2-intermediate/015-bookmyshow/boilerplate/cpp/part3/guided.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>
using namespace std;


// Data class (given).
struct ShowOp {
    string kind;
    string s1;
    string s2;
    string s3;
    string s4;
    int i1;
    int i2;
    int i3;
    ShowOp(const string& kind_, const string& s1_ = "", const string& s2_ = "", const string& s3_ = "", const string& s4_ = "", int i1_ = 0, int i2_ = 0, int i3_ = 0)
      : kind(kind_), s1(s1_), s2(s2_), s3(s3_), s4(s4_), i1(i1_), i2(i2_), i3(i3_) {}
};

// HINT: introduce an abstraction so new variants don't change existing code.
// HINT: keep each piece small — one responsibility per class.

// HINT: start from what this must return, then work backwards to the state it needs.
vector<string> show_simulate(vector<ShowOp> ops) {
    // TODO: write your solution
    return {};
}

File: problems/tier2-intermediate/019-auction-system/boilerplate/java/part2/Guided.java
import java.util.*;

// Data class (given).
class AuctionOp {
    public String kind;
    public String s1;
    public String s2;
    public String s3;
    public int i1;
    public int i2;
    public int i3;

    public AuctionOp(String kind, String s1, String s2, String s3, int i1, int i2, int i3) {
        this.kind = kind;
        this.s1 = s1;
        this.s2 = s2;
        this.s3 = s3;
        this.i1 = i1;
        this.i2 = i2;
        this.i3 = i3;
    }

    public AuctionOp(String kind) {
        this(kind, "", "", "", 0, 0, 0);
    }
}

// HINT: introduce an abstraction so new variants don't change existing code.
public class Solution {
    // HINT: start from what this must return, then work backwards to the state it needs.
    public static List<String> auction_simulate(List<AuctionOp> ops) {
        // TODO: write your solution
        return null;
    }

}
